pilot/lib/pilot/build_manager.rb

```diff
@@ -18,18 +18,27 @@ def upload(options)
       transporter = FastlaneCore::ItunesTransporter.new(options[:username])
       result = transporter.upload(app.apple_id, package_path)
 
-      if result
-        UI.message("Successfully uploaded the new binary to iTunes Connect")
+      unless result
+        UI.user_error!("Error uploading ipa file, more information see above")
+      end
 
-        unless config[:skip_submission]
-          uploaded_build = wait_for_processing_build
-          distribute_build(uploaded_build, options)
+      UI.message("Successfully uploaded the new binary to iTunes Connect")
 
-          UI.message("Successfully distributed build to beta testers 🚀")
-        end
-      else
-        UI.user_error!("Error uploading ipa file, more information see above")
+      if config[:skip_waiting_for_build_processing]
+        UI.important("Skip waiting for build processing")
+        UI.important("This means, no changelog will be set and no build will be distributed to testers")
+        return
       end
+
+      UI.message("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option")
+      uploaded_build = wait_for_processing_build # this might take a while
+      # First, set the changelog (if necessary)
+      uploaded_build.update_build_information!(whats_new: options[:changelog])
+      UI.success "Successfully set the changelog for build"
```

Comment:
Do we want to do this/show this message if the user hasn't provided a changelog/what-to-test?